### PR TITLE
LOAP-3267 | Limit amount of requests

### PR DESCRIPTION
**Pull request by kzadurskakreditech**
### Purpose
Right now we send too many requests to dadata.

### Approach
* Update `componentDidUpdate` so that it has the same rules as input on change handler. 
* Don't send request with empty query (e.g. backspacing)
* Send request after a certain number of characters is entered

### Extra
* Rename `transpile` to `build` - it is more aligned with our other repos.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -153,6 +153,8 @@ interface BaseInputProps<T = HTMLInputElement> {
  * @property  { boolean } allowClear - [optional] show/hide clear fieldd control
  * @property  { boolean } showNote - [optional] show/hide note at suggestions list
  * @property  { boolean } silentQuery - [optional] initial query that not showed on input but determine  suggestion list that showed at first
+ * @property  { string } name - [optional] - property name of input (default: "daData")
+ * @property  { number } minimumCharacterThreshold - [optional] minimum number of characters entered after which a first request is sent (default: 3)
```
Comment (marzena-malachowska-kreditech) [inline index.d.ts:158]: I didn't notice this file when I was making some changes in this repo. 
Could you add one more prop that I added before (https://github.com/kzadurskakreditech/react-dadata-box/commit/33d51a9f11a0d310636fdfc9fce990b10f9ef559) 🙏
`autoFocus `

*@property { boolean } autoFocus - [optional] add autofocus to the field

**Review by marzena-malachowska-kreditech — APPROVED** [review]
🚀

File: index.d.ts (hunk 1 above)
Comment (kzadurskakreditech) [reply index.d.ts:158]: 9e46878 ✅

Comment (robert-kirsz-kreditech) [reply index.d.ts:158]: This whole file seems cool, we should think about adopting it in CCL

**Review by robert-kirsz-kreditech — APPROVED** [review]
🌮

**Pull request merged**